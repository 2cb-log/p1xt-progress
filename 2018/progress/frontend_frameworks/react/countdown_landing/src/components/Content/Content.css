.Content {
    text-align: left;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin: 100px;
}

.Content__banner {
    font-family: 'Dosis', sans-serif;
    font-weight: bold;
    width: 50%;
}

.Content__details {
    font-family: 'Dosis', sans-serif;
    width: 50%;
}

.Content__notes {
    font-family: 'Roboto', sans-serif;
}