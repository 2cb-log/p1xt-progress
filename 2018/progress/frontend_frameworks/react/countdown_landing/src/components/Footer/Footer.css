.Footer {
    text-align: left;
    background: #202020;
}

.Footer__content {
    color: #858585;
    padding-top: 20px;
    padding-bottom: 20px;
}

.Footer__content a,
.Footer__content a:visited {
    color: #797979;
}

.Footer__content a:hover {
    color: #b1b1b1;
}

.Footer__icons img {
    width: 25px;
    height: 25px;
    padding-right: 10px;
}