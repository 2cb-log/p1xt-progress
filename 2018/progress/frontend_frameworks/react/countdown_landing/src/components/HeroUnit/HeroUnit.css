.HeroUnit {
    text-align: center;
    display: flex;
    flex-direction: column;
    background: url(/assets/images/benjamin-voros-250200.jpg);
    background-size: 100vw;
    min-height: 500px;
    width: 100vw;
}

.HeroUnit__transparency {
    position: absolute;
    top: 0;
    left: 0;
    background: #272727;
    opacity: .8;
    height: 500px;
    width: 100vw;
}