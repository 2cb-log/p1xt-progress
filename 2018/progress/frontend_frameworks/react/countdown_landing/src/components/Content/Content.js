import React from 'react';
import './Content.css';

function Content(props) {
  return (
    <div className="Content">
      Photo by Benjamin Voros on Unsplash
    </div>
  );

}

export default Content;
