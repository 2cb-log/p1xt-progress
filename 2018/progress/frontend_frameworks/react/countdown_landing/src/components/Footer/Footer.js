import React from 'react';
import './Footer.css';

function Footer(props) {
  return (
    <div className="Footer">
      <div className="Footer__content container">
        <p>Created by P1xt - inspired by <a href="http://www.free-css.com/free-css-templates/page221/orly">Designsub</a>
        </p>
        <p className="Footer__icons">
          <a href="https://github.com/P1xt">
            <img src="assets/icons/github.svg" alt="Github" />
          </a>
          <a href="https://twitter.com/P1xt">
            <img src="assets/icons/twitter.svg" alt="Github" />
          </a>
        </p>
      </div>
    </div>
  );

}

export default Footer;
