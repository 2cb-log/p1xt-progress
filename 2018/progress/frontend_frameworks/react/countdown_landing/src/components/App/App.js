import React from 'react';
import HeroUnit from '../HeroUnit';
import Content from '../Content';
import Footer from '../Footer';
import Subscribe from '../Subscribe';

import './App.css';

function App(props) {
  return (
    <div className="App">
      <HeroUnit />
      <Content />
      <Subscribe />
      <Footer />
    </div>
  );

}

export default App;
