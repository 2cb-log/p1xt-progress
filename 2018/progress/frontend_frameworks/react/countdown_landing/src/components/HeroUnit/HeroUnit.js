import React from 'react';
import './HeroUnit.css';
import Logo from '../Logo';
import Countdown from '../Countdown';
import NotifyButton from '../NotifyButton';

function HeroUnit(props) {
  return (
    <div className="HeroUnit">
      <div className="HeroUnit__transparency"></div>
      <Logo />
      <Countdown />
      <NotifyButton />
    </div>
  );
}

export default HeroUnit;
