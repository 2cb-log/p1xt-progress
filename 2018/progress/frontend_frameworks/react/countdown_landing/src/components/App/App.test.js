import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import HeroUnit from '../HeroUnit';
import Content from '../Content';
import Footer from '../Footer';

describe('<App />', () => {
  const wrapper = shallow(<App />);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders HeroUnit', () => {
    expect(wrapper.find(HeroUnit).length).toBe(1);
  });

  it('renders Content', () => {
    expect(wrapper.find(Content).length).toBe(1);
  });

  it('renders Footer', () => {
    expect(wrapper.find(Footer).length).toBe(1);
  });
});

