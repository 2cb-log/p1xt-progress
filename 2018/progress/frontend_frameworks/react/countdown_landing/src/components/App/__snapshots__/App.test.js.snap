// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="HeroUnit"
  >
    <div
      className="HeroUnit__transparency"
    />
    <div
      className="Logo"
    >
      Logo
    </div>
    <div
      className="Countdown"
    >
      Countdown
    </div>
    <div
      className="NotifyButton"
    >
      NotifyButton
    </div>
  </div>
  <div
    className="Content"
  >
    Photo by Benjamin Voros on Unsplash
  </div>
  <div
    className="Footer"
  >
    Footer
  </div>
</div>
`;
